zingolib/src/wallet/transaction_record.rs

```diff
@@ -0,0 +1,343 @@
+use incrementalmerkletree::witness::IncrementalWitness;
+use zcash_primitives::{sapling, transaction::TxId};
+
+use crate::error::ZingoLibError;
+
+use super::{
+    data::{OutgoingTxData, PoolNullifier},
+    *,
+};
+
+///  Everything (SOMETHING) about a transaction
+#[derive(Debug)]
+pub struct TransactionRecord {
+    // the relationship of the transaction to the blockchain. can be either Broadcast (to mempool}, or Confirmed.
+    pub status: ConfirmationStatus,
+
+    // Timestamp of Tx. Added in v4
```

Comment:
I've always wondered how reliable this timestamp is. Is it actually from the transaction, or the block that mined it? If from the transaction, do the consensus rules guarantee that it falls within a reasonable time frame relative to the block? I think it would be good to clarify some of this in the docs.